ehrql/query_language.py

```diff
@@ -1531,6 +1498,63 @@ def minimum_of(value, other_value, *other_values):
 #
 
 
+def raise_helpful_error_if_possible(arg):
+    if isinstance(arg, BaseFrame):
+        raise TypeError(
+            f"Expecting a series but got a frame (`{arg.__class__.__name__}`): "
+            f"are you missing a column name?"
+        )
+    if callable(arg):
+        raise TypeError(
+            f"Function referenced but not called: are you missing parentheses on "
+            f"`{arg.__name__}()`?"
+        )
+
+
+def validate_ehrql_series(name, arg):
+    try:
+        raise_helpful_error_if_possible(arg)
+    except TypeError as e:
+        raise TypeError(f"invalid {name}:\n{e})") from None
+    if not isinstance(arg, BaseSeries):
+        raise TypeError(
+            f"invalid {name}:\n"
+            f"Expecting an ehrQL series, got type '{type(arg).__qualname__}'"
+        )
+
+
+def validate_patient_series(name, arg):
```

Comment:
I found the actual values for `name` passed to this function (and the other similar `validate_...()` functions) slightly confusing (e.g. "f"`group_by` value for '{key}'"). Here I'm expecting it to be the name of a thing, and it's not really, it's more like a description. I'm not sure `description` is a better name for the argument though...perhaps just using the keyword when it's called would be enough?

Reply:
You're quite right, thanks. I've changed the name to `context` and passed it as keyword arg instead of the first arg. I think it's much clearer now. Let me know if you agree.

Reply:
Much better, thanks :)